### PR TITLE
Modified route_guide sample app to show RPC log

### PR DESCRIPTION
**Pull request by makdharma**
Modified the screens to show meaningful information about RPC progress.
When you click on each tab the screen now shows log messages that previously
only went to the debug console.

### CODE REVIEW

**Comment by jcanizales** [comment]
Please use camelCase for names

File: examples/objective-c/route_guide/ViewControllers.m [hunk 1]
```diff
@@ -37,7 +37,7 @@
 #import <RxLibrary/GRXWriter+Immediate.h>
 #import <RxLibrary/GRXWriter+Transformations.h>
 
-static NSString * const kHostAddress = @"localhost:50051";
+static NSString * const kHostAddress = @"192.168.1.120:50051";
```
Comment (jcanizales) [inline examples/objective-c/route_guide/ViewControllers.m]: ? :)

**Comment by makdharma** [comment]
@jcanizales The test failures are unrelated since we don't build this sample. Anything else needs doing?

File: examples/objective-c/route_guide/ViewControllers.m [hunk 2]
```diff
@@ -83,17 +83,24 @@ + (instancetype)noteWithMessage:(NSString *)message
 @interface GetFeatureViewController : UIViewController {
   RTGRouteGuide *service;
```
Comment (jcanizales) [inline examples/objective-c/route_guide/ViewControllers.m]: I missed this when reviewing #6440: ivars should be declared in the `@implementation`, and named with a `_` prefix (per https://google.github.io/styleguide/objcguide.xml#Instance_Variables_In_Headers_Should_Be_@private ). I think the best here for clarity would be to just turn it into a `@property`.

File: examples/objective-c/route_guide/ViewControllers.m [hunk 3]
```diff
@@ -83,17 +83,24 @@ + (instancetype)noteWithMessage:(NSString *)message
...
 @end
 
 @implementation GetFeatureViewController
 
 - (void)execRequest {
   void (^handler)(RTGFeature *response, NSError *error) = ^(RTGFeature *response, NSError *error) {
     if (response.name.length) {
+      NSString *str =[NSString stringWithFormat:@"%@\nFound feature called %@ at %@.", self.outputLabel.text, response.location, response.name];
+      self.outputLabel.text = str;
       NSLog(@"Found feature called %@ at %@.", response.name, response.location);
```
Comment (jcanizales) [inline examples/objective-c/route_guide/ViewControllers.m]: It looks like we could remove boilerplate for the people reading the example by doing:
```objective-c
if (response.name.length) {
  [self log:@"Found feature called %@ at %@.", response.name, response.location];
} ...
```
The log method can be declared, along with the `outputLabel` and `service` properties, in a superclass that's common to all these view controllers:
```objc
@interface SampleViewController : UIViewController
@property(weak, nonatomic) IBOutlet UILabel *outputLabel;
@property(strong, nonatomic) RTGRouteGuide *service;

- (void)log:(NSString *)formatString, ...;
@end

@implementation SampleViewController
- (void)log:(NSString *)formatString, ... {
  va_list args;
  va_start(args, formatString);
  NSString *str = [NSString stringWithFormat:formatString arguments:args];
  self.outputLabel.text = [self.outputLabel.text stringByAppendingString:str];
  NSLog(str);
  va_end(args);
}
@end
```

File: examples/objective-c/route_guide/ViewControllers.m [hunk 4]
```diff
@@ -116,6 +123,9 @@ - (void)viewDidLoad {
 }
 
 - (void)viewDidAppear:(BOOL)animated {
+  self.outputLabel.text = @"RPC log:";
+  self.outputLabel.numberOfLines = 0;
+  self.outputLabel.font = [UIFont fontWithName:@"Helvetica Neue" size:8.0];
```
Comment (jcanizales) [inline examples/objective-c/route_guide/ViewControllers.m]: I think we should be able to set these two properties via the UI builder, to get them out of the way too.

**Comment by jcanizales** [comment]
Just one style issue, and two ideas to remove boilerplate.

File: examples/objective-c/route_guide/ViewControllers.m [hunk 5]
```diff
@@ -81,7 +81,7 @@ + (instancetype)noteWithMessage:(NSString *)message
  * not to have a feature.
  */
 @interface GetFeatureViewController : UIViewController {
-  RTGRouteGuide *service;
+  RTGRouteGuide *_service;
```
Comment (jcanizales) [inline examples/objective-c/route_guide/ViewControllers.m]: And declared in the implementation, not in the interface :)

File: examples/objective-c/route_guide/ViewControllers.m (hunk 4 above)
Comment (jcanizales) [inline examples/objective-c/route_guide/ViewControllers.m]: TODO for this too?

**Comment by jcanizales** [comment]
LGTM!

**Comment by makdharma** [comment]
test this please

**Comment by makdharma** [comment]
test this please

**Pull request merged**